Training24/Test.c

```diff
@@ -1,19 +1,187 @@
+      printf ("\n\t+---------------------+--------------------------+-------------------+-------------------+--------+\n"
+         "\t| INTEGER - TEST CASE |     REVERSED INTEGER     |  EXPECTED OUTPUT  |   ACTUAL OUTPUT   | RESULT |\n"
+         "\t+---------------------+--------------------------+-------------------+-------------------+--------+\n");
+      break;
+   default:
+      printf ("\tInvalid header type.\n"); break;
+   }
+}
+
+void PhraseTableRow (const char* input, const char* expectedOutput, const char* actualOutput, const char* result) {
+   printf ("\t| %-20s | %-17s | %-17s | %-15s |\n", input, expectedOutput, actualOutput, result);
+   printf ("\t+----------------------+-------------------+-------------------+--------+\n");
+}
+
+void IntegerTableRow (int number, const char* outputStr, const char* actualOutput, const char* expectedOutput) {
+   const char* resultStr = (strcmp (actualOutput, expectedOutput) == 0) ? GREEN "PASS" RESET : RED "FAIL" RESET;
+   printf ("\t| %-19d | %-24s | %-17s | %-17s | %-15s |\n", number, outputStr, actualOutput, expectedOutput, resultStr);
+   printf ("\t+---------------------+--------------------------+-------------------+-------------------+--------+\n");
+}
+
+void TestPhrasePalindromes () {
+   PrintTableHeader (Phrase);
+   const char* phrases[] = { "tacOcAT", "hello", "D", "Don't nod", "Race!Car", " ", "22 / 02 / 2022",
+      "12345678987654321", "0123456789" },
+      * expectedResults[] = { "Palindrome", "Not a Palindrome", "Palindrome", "Palindrome",
+      "Palindrome", "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome" };
+   for (int i = 0; i < sizeof (phrases) / sizeof (phrases[0]); i++) {
+      int isPhrasePalindrome = CheckPhrasePalindrome (phrases[i]);
+      const char* actualResult = isPhrasePalindrome ? "Palindrome" : "Not a Palindrome",
+         * output = (strcmp (actualResult, expectedResults[i]) == 0) ? GREEN "PASS" RESET : RED "FAIL" RESET;
+      PhraseTableRow (phrases[i], expectedResults[i], actualResult, output);
+   }
+   printf ("\n");
+}
+
+void TestReverseNumbers () {
+   int testCases[] = { 212, -121, 12345, -9876789, 999999999, 2147447412, 2147483647, -2147483648LL }, reversedNumber;
+   const char* expectedOutputs[] = { "Palindrome", "Not a Palindrome", "Not a Palindrome",
+      "Not a Palindrome", "Palindrome", "Palindrome", "Failed to process", "Not a Palindrome" };
+   PrintTableHeader (Integer);
+   for (int i = 0; i < sizeof (testCases) / sizeof (testCases[0]); i++) {
+      int number = testCases[i];
+      char outputStr[256];
+      bool overflow = false;
+      if (number < 0) IntegerTableRow (number, "Invalid(Negative number)", "Not a Palindrome", expectedOutputs[i]);
+      else {
+         overflow = ReverseAndCheckOverflow (number, &reversedNumber);
+         if (overflow) {
+            strcpy (outputStr, "Integer Overflow");
+            IntegerTableRow (number, outputStr, "Failed to process", expectedOutputs[i]);
```

Comment:
assign failed to process as actual output if overflow is detected and print like this.
```suggestion
            IntegerTableRow (number, outputStr, actualOutput, expectedOutputs[i]);
```